pkg/upgrade/upgrade.go

```diff
@@ -175,44 +182,47 @@ func (ci *clusterInfo) updateInfo(info map[string]NodeVersion, dtk registry.Driv
 }
 
 func (ci *clusterInfo) driverToolkitVersion(ctx context.Context, entries []string, info map[string]NodeVersion) (map[string]NodeVersion, error) {
-
 	for _, entry := range entries {
-
-		ci.log.Info("History", "entry", entry)
-
-		var (
-			err   error
-			layer v1.Layer
-		)
-
-		layer, err = ci.registry.LastLayer(ctx, entry)
-		if err != nil {
-			return nil, err
-		}
-
-		if layer == nil {
-			continue
-		}
-		// For each entry we're fetching the cluster version and dtk URL
-		_, imageURL, err := ci.registry.ReleaseManifests(layer)
-		if err != nil {
-			return nil, fmt.Errorf("could not extract version from payload: %w", err)
-		}
-
-		if imageURL == "" {
-			utils.WarnOnError(errors.New("No DTK image found, DTK cannot be used in a Build"))
-			return info, nil
-		}
-
-		if layer, err = ci.registry.LastLayer(ctx, imageURL); layer == nil {
-			return nil, fmt.Errorf("cannot extract last layer for DTK from %s: %w", imageURL, err)
-		}
-
-		dtk, err := ci.registry.ExtractToolkitRelease(layer)
-		if err != nil {
-			return nil, err
+		var dtk registry.DriverToolkitEntry
+		var imageURL string
+		if cached, ok := ci.cache[entry]; ok {
```

Comment:
If we switch to `*cacheEntry` values in the map, we can do:
```suggestion
		if cached := ci.cache[entry]; cached != nil {
```